main.py

```diff
@@ -19,127 +26,117 @@
+            .field("forwarded", int(pihole.forwarded.replace(',',''))) \
+            .field("cached", int(pihole.cached.replace(',',''))),
+
+        Point("queries") \
+            .time(timestamp) \
+            .tag("hostname", PIHOLE_HOSTNAME) \
+            .field("queries", int(pihole.queries.replace(',',''))) \
+            .field("blocked", int(pihole.blocked.replace(',',''))) \
+            .field("ads_percentage", float(pihole.ads_percentage)),
+
+        Point("clients") \
+            .time(timestamp) \
+            .tag("hostname", PIHOLE_HOSTNAME) \
+            .field("total_clients", int(pihole.total_clients.replace(',',''))) \
+            .field("unique_clients", int(pihole.unique_clients.replace(',',''))) \
+            .field("total_queries", int(pihole.total_queries.replace(',',''))),
+
+        Point("other") \
+            .time(timestamp) \
+            .tag("hostname", PIHOLE_HOSTNAME) \
+            .field("status", pihole.status == 'enabled') \
+            .field("gravity_last_update", pihole.gravity_last_updated['absolute'])
     ]
 
-    return influx_data
+def get_authenticated_data_for_influxdb(pihole: PiHole, timestamp: datetime.datetime):
+    query_type_point = Point("query_types") \
+        .time(timestamp) \
+        .tag("hostname", PIHOLE_HOSTNAME)
+    
+    for key, value in pihole.query_types.items():
+        query_type_point.field(key, float(value))
 
-def get_formatted_authenticated_query_types():
-    formatted_dict = {}
-    for key in pihole.query_types:
-        formatted_dict[key] = float(pihole.query_types[key])
+    forward_destinations_point = Point("forward_destinations") \
+        .time(timestamp) \
+        .tag("hostname", PIHOLE_HOSTNAME)
     
-    return formatted_dict
-
-def get_formatted_authenticated_forward_destinations():
-    formatted_dict = {}
-    for key in pihole.forward_destinations['forward_destinations']:
-        formatted_dict[key.split('|')[0]] = pihole.forward_destinations['forward_destinations'][key]
-    return formatted_dict
-
-def get_authenticated_data_for_influxdb():
-    influx_data = [
-        {
-            'measurement': 'authenticated_query_types',
-            'time': datetime.datetime.now(),
-            'fields': get_formatted_authenticated_query_types()
-        },
-        {
-            'measurement': 'authenticated_forward_destinations',
-            'time': datetime.datetime.now(),
-            'fields': get_formatted_authenticated_forward_destinations()
-        }
+    for key, value in pihole.forward_destinations['forward_destinations'].items():
+        forward_destinations_point.field(key.split('|')[0], value)
+
+    return [
+        query_type_point,
+        forward_destinations_point
     ]
 
-    return influx_data
+class Auth(object):
+    def __init__(self, token):
+        # PiHole's web token is just a double sha256 hash of the utf8 encoded password
+        self.token = token
+        self.auth_timestamp = time.time()
 
 def main():
```

Comment:
Could we add a check here to see if someone is using the old version and log a warning.